django_mongodb/utils.py

```diff
@@ -18,3 +24,89 @@ def check_django_compatability():
             f"You must use the latest version of django-mongodb {A}.{B}.x "
             f"with Django {A}.{B}.y (found django-mongodb {__version__})."
         )
+
+
+class CollectionDebugWrapper:
+    def __init__(self, collection, db):
+        self.collection = collection
+        self.db = db
+
+    def __getattr__(self, attr):
+        return getattr(self.collection, attr)
+
+    def profile_call(self, func, args=(), kwargs=None):
+        start = time.monotonic()
+        retval = func(*args, **kwargs or {})
+        duration = time.monotonic() - start
+        return duration, retval
+
+    def log(self, op, duration, args, kwargs=None):
+        msg = "(%.3f) %s"
+        args = " ".join(str(arg) for arg in args)
+        operation = f"{self.collection.name}.{op}({args})"
+        kwargs = {k: v for k, v in kwargs.items() if v}
+        if kwargs:
+            operation += f"; kwargs={kwargs}"
+        if len(settings.DATABASES) > 1:
+            msg += f"; alias={self.db.alias}"
+        self.db.queries_log.append(
+            {
+                "sql": operation,
+                "time": "%.3f" % duration,
+            }
+        )
+        logger.debug(
+            msg,
+            duration,
+            operation,
+            extra={
+                "duration": duration,
+                "sql": operation,
+                "kwargs": kwargs,
+                "alias": self.db.alias,
+            },
+        )
+
+    def find(self, *args, **kwargs):
+        return DebugCursor(self, self.collection, *args, **kwargs)
+
+    def logging_wrapper(method):
+        def wrapper(self, *args, **kwargs):
+            func = getattr(self.collection, method)
+            # Collection.insert_one() mutates args[0] (the document) by adding
+            # the _id. deepcopy() to avoid logging that version.
+            original_args = copy.deepcopy(args)
```

Comment:
Maybe this `deepcopy` should be omitted. I just came across this [in the MongoDB docs](https://pymongo.readthedocs.io/en/stable/faq.html#writes-and-ids): "Copying the document to insert before adding the _id field would be prohibitively expensive for most high write volume applications."

It was surprising to me to see queries logged as if the user has supplied an _id, but perhaps it's okay.